index.html

```diff
@@ -4820,6 +4820,113 @@ <h3>Verification</h3>
+          </li>
+          <li>
+zero or more errors ([=list=] of [=ProblemDetails=] |errors|)
+          </li>
+        </ul>
+
+        <p>
+The context validation algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set |result|.|status| to `false`.
+          </li>
+          <li>
+For each |contextValue| expressed in every `@context` property, ensure that
+either:
+            <ol class="algorithm">
+              <li>
+The |contextValue| exists in |approvedContexts| and matches a cryptographic
+hash associated with the matching approved context, or
+              </li>
+              <li>
+optionally, for the context values that do not exist in |approvedContexts|,
+remove the |contextValue| from the `@context` property values. Set
+|result|.|document| to the result of running the
+<a data-cite="JSON-LD11-API#compaction-algorithm">
+JSON-LD Compaction Algorithm</a> with the modified |inputDocument| as input.
+              </li>
+            </ol>
+          </li>
+          <li>
+If no errors occurred in the previous step, set |result|.|status| to `true` and,
+if |result|.|document| is not set, set it to |inputDocument|. If errors
+occurred, set |result|.|status| to `false`, remove the |document| property from
+|result|, and add at least one
+<a href="#MALFORMED_VALUE_ERROR">MALFORMED_VALUE_ERROR</a> to |result|.|errors|.
+          </li>
+          <li>
+Return the value of |result|.
+          </li>
+        </ol>
+
+        <p>
+The purpose of the algorithm above is to ensure that the [=verifier=] has
+explicitly approved of the types of [=verifiable credentials=] and [=verifiable
+presentations=] that it will process. Not checking these JSON-LD Context values
+against known good values, and the semantics that they convey, can lead to
+security vulnerabilities. While the algorithm above provides one way of checking
+these values, and one optional way of safely processing unknown values, there
+are other approaches that provide the same protections.
+        </p>
+
+        <p>
+One such approach would be for a [=verifier=] to configure a JSON-LD Context
+loader, sometimes referred to as a document loader, to use only local copies of
+approved context files which would guarantee that 1) context files will never
+change, and 2) the cryptographic hashes will never change, effectively
+resulting in the same result as the algorithm above.
+        </p>
+
+        <p>
+Another approach a [=verifier=] could use would be to use a list of well known
+context URLs and their associated approved cryptographic hashes, without storing
+every context file locally, which would allow contexts to be safely loaded and
+cached from the network without compromising the security expectations of the
+[=verifier=]. This approach would also be effectively equivalent to the
+algorithm provided in this section.
+        </p>
+
+        <p>
+Yet another valid approach would be for the [=holder=] to
```

Comment:
Listing multiple ways to perform validation is good, but it then seems that the above MUST on the specific algorithm provided there should be a MAY instead -- or some other text should make it clear that either the algorithm can be run or one of these other equivalent approaches can be taken.

Reply:
Fixed in 8dbad0b15b3b4d2ad46f6ba381951b6d73ca8990.